### PR TITLE
Minor cleanup to Highlighter types

### PR DESCRIPTION
**Pull request by jjatie**

### CODE REVIEW

File: Source/Charts/Highlight/ChartHighlighter.swift [hunk 1]
```diff
@@ -48,19 +48,15 @@ open class ChartHighlighter : NSObject, IHighlighter
...
-        guard let chart = chart
-            else { return nil }
+        guard let chart = chart else { return nil }
         
         let closestValues = getHighlights(xValue: xVal, x: x, y: y)
-        if closestValues.isEmpty
-        {
-            return nil
-        }
+        guard !closestValues.isEmpty else { return nil }
```
Comment (petester42) [inline Source/Charts/Highlight/ChartHighlighter.swift]: This guard is harder to read than how it was before. I have to think about it for a second when I try to understand this line. It was more straightforward before. It doesn't reduce nesting so we can switch it back.

File: Source/Charts/Highlight/PieHighlighter.swift [hunk 2]
```diff
@@ -17,12 +17,10 @@ open class PieHighlighter: PieRadarHighlighter
 {
     open override func closestHighlight(index: Int, x: CGFloat, y: CGFloat) -> Highlight?
     {
-        guard let set = chart?.data?.dataSets[0]
+        guard let set = chart?.data?.dataSets[0],
```
Comment (petester42) [inline Source/Charts/Highlight/PieHighlighter.swift]: Combining these guard statements makes it harder to read. I think it was easier to understand before

File: Source/Charts/Highlight/PieRadarHighlighter.swift [hunk 3]
```diff
@@ -23,32 +23,30 @@ open class PieRadarHighlighter: ChartHighlighter
         let touchDistanceToCenter = chart.distanceToCenter(x: x, y: y)
         
         // check if a slice was touched
-        if touchDistanceToCenter > chart.radius
-        {
+        guard touchDistanceToCenter <= chart.radius else {
```
Comment (petester42) [inline Source/Charts/Highlight/PieRadarHighlighter.swift]: guard or if here doesn't really make a difference.

File: Source/Charts/Highlight/ChartHighlighter.swift (hunk 1 above)
Comment (jjatie) [reply Source/Charts/Highlight/ChartHighlighter.swift]: While `guard` often does reduce nesting, it's purpose is to indicate that
> You use a guard statement to require that a condition must be true in order for the code after the guard statement to be executed. [Apple](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ControlFlow.html).

It seems reasonable to me to expect reading "only continue if `closestValues` is not empty"

File: Source/Charts/Highlight/PieHighlighter.swift (hunk 2 above)
Comment (jjatie) [reply Source/Charts/Highlight/PieHighlighter.swift]: What about all the other places in the framework where we are writing `guard` statements like this? Although usually it's written like 
```
guard
    let set = chart?.data?.dataSets[0],
    let entry = set.entryForIndex(index)
    else { return nil }
```

File: Source/Charts/Highlight/PieRadarHighlighter.swift (hunk 3 above)
Comment (jjatie) [reply Source/Charts/Highlight/PieRadarHighlighter.swift]: Again:
> You use a guard statement to require that a condition must be true in order for the code after the guard statement to be executed.

So in this case the guard makes it extra clear that the method can return early here

**Comment by jjatie** [comment]
The convention is as the [Swift Book](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ControlFlow.html) states:
> You use a guard statement to require that a condition must be true in order for the code after the guard statement to be executed.

While technically it doesn't matter if we use an `if` or a `guard`, the `guard` does make it clear that we need a condition to be `true` to continue.

As for the longer `guard` conditions I don't know of any convention, but I like the way it's written in many places in Charts and I will update this one to look the same. (e.g.
```
        guard
            let data = _data as? BarChartData,
            let set = data.getDataSetForEntry(e) as? IBarChartDataSet
            else { return .null }
```

**Comment by jjatie** [comment]
@petester42 Do you have any further feedback or can this be merged?

**Comment by jjatie** [comment]
Review has been discussed in #3034 and approved.

**Comment by liuxuan30** [comment]
next time merging some old PRs, just merge master and trigger a new CI to make sure breaks nothing :)

**Comment by jjatie** [comment]
I did : )

**Pull request merged**